### PR TITLE
Fix defects being mandatory

### PR DESCRIPTION
**Pull request by mfisherlevine**

### CODE REVIEW

File: python/lsst/ip/isr/isrTask.py [hunk 1]
```diff
@@ -399,8 +403,8 @@ def run(self, ccdExposure, bias=None, linearizer=None, dark=None, flat=None, def
             fringes = pipeBase.Struct(fringes=None)
         if self.config.doFringe and not isinstance(fringes, pipeBase.Struct):
             raise RuntimeError("Must supply fringe exposure as a pipeBase.Struct")
-
-        defects = [] if defects is None else defects
+        if self.config.doDefect and defects is None:
+            raise RuntimeError("Must supply defects if config.doDefect True")
```
Comment (RobertLuptonTheGood) [inline python/lsst/ip/isr/isrTask.py]: I'd use a verb...

File: python/lsst/ip/isr/isrTask.py [hunk 2]
```diff
@@ -64,6 +64,11 @@ class IsrTaskConfig(pexConfig.Config):
         doc="Apply fringe correction?",
         default=True,
     )
+    doDefect = pexConfig.Field(
+        dtype=bool,
+        doc="Apply defect correction?",
```
Comment (RobertLuptonTheGood) [inline python/lsst/ip/isr/isrTask.py]: "Correct CCD defects such as hot pixels"?

File: python/lsst/ip/isr/isrTask.py (hunk 1 above)
Comment (mfisherlevine) [inline python/lsst/ip/isr/isrTask.py]: So would I, but none of other error messages there do, so I follow form.

File: python/lsst/ip/isr/isrTask.py (hunk 2 above)
Comment (mfisherlevine) [inline python/lsst/ip/isr/isrTask.py]: These are all very short, and if you don't know what CCD defects are then I don't think this is the place to learn.

Comment (mfisherlevine) [inline python/lsst/ip/isr/isrTask.py]: OK, done.

**Pull request merged**